File: API/API/repositorys/RefreshTokenRepository.vb
Public Class RefreshTokenRepository
    Implements IRefreshTokenRepository

    Private ReadOnly _db As AppDbContext

    Public Sub New(db As AppDbContext)
        _db = db
    End Sub

    Public Function GetByToken(token As String) As RefreshToken _
        Implements IRefreshTokenRepository.GetByToken

        Return _db.RefreshTokens.
            FirstOrDefault(Function(x) x.Token = token)

    End Function

    Public Function GetActiveTokenByUserId(userId As Guid) As RefreshToken _
        Implements IRefreshTokenRepository.GetActiveTokenByUserId

        Return _db.RefreshTokens.FirstOrDefault(
        Function(x) x.UserId = userId AndAlso
                    x.RevokedAt Is Nothing AndAlso
                    x.ExpiresAt > DateTime.UtcNow)

    End Function

    Public Sub Add(token As RefreshToken) _
        Implements IRefreshTokenRepository.Add

        _db.RefreshTokens.Add(token)

    End Sub

    Public Sub Save() _
        Implements IRefreshTokenRepository.Save

        _db.SaveChanges()

    End Sub

End Class

File: API_v2/Repositorys/RefreshTokenRepository.cs
using API_v2.Datas;
using API_v2.Models;
using API_v2.Repositorys.IRepositorys;

namespace API_v2.Repositorys
{
    public class RefreshTokenRepository : IRefreshTokenRepository
    {
        private readonly AppDbContext _db;

        public RefreshTokenRepository(AppDbContext db)
        {
            _db = db;
        }

        public RefreshToken? GetByToken(string token)
        {
            return _db.RefreshTokens.FirstOrDefault(x => x.Token == token);
        }

        public RefreshToken? GetActiveTokenByUserId(Guid userId)
        {
            return _db.RefreshTokens.FirstOrDefault(x => x.UserId == userId && x.RevokedAt == null && x.ExpiresAt > DateTime.UtcNow);
        }

        public List<RefreshToken> GetActiveTokensByUserId(Guid userId)
        {
            return _db.RefreshTokens.Where(x => x.UserId == userId && x.RevokedAt == null && x.ExpiresAt > DateTime.UtcNow).ToList();
        }

        public void Add(RefreshToken token)
        {
            _db.RefreshTokens.Add(token);
        }

        public void Save()
        {
            _db.SaveChanges();
        }
    }
}
